### ISSUE TITLE
Unable to upload videos, 502 bad gateway

### ISSUE THREAD

**Issue by Powersource** [download13/ipfstube#6]
Specifically [Sintel](magnet:?xt=urn:btih:6a9759bffd5c0af65319979fb7832189f4f3c35d&dn=sintel.mp4&tr=udp%3a%2f%2ftracker.openbittorrent.com%3a80&tr=udp%3a%2f%2ftracker.internetwarriors.net%3a1337&tr=udp%3a%2f%2ftracker.leechers-paradise.org%3a6969&tr=udp%3a%2f%2ftracker.coppersurfer.tk%3a6969&tr=udp%3a%2f%2fexodus.desync.com%3a6969&tr=wss%3a%2f%2ftracker.openwebtorrent.com&tr=wss%3a%2f%2ftracker.btorrent.xyz&ws=https%3a%2f%2fwebtorrent.io%2ftorrents%2fsintel-1024-surround.mp4) but I've tried another one too (.mp4 as well). Tried Firefox and Chrome.

**Comment by Powersource** [comment]
Ooooh I just realized you're running everything through a few gateways (if I'm not misunderstanding). Should probably move to/reimplement with https://github.com/ipfs/js-ipfs

**Comment by download13** [comment]
Yeah, it's just based on the local gateway, falling back to others if it crashes (or is just frozen after filling its entire drive). I was under the impression that DHT support wasn't fully implemented yet for js-ipfs. Does it actually work?
